### ISSUE TITLE
Ruby Error on view template preview screen?

### ISSUE THREAD

**Issue by dsenecalb** [gocd/gocd#1585]
Hi all, 

Creating a pipeline, I was trying to link it into one of the templates. While Trying to see the preview of the template I saw the following error:

![gocd error](https://cloud.githubusercontent.com/assets/12563338/10829777/74dfd9c4-7e52-11e5-91ca-d5a4831ad00c.png)

And looking from the log, I seen this Ruby error. What does it means?

```
2015-10-29 13:34:27,876 FATAL [qtp2133131964-31] Rails:? -
ActionView::Template::Error (undefined method `workingDirectory' for #<Java::ComThoughtworksGoConfigPluggabletask::PluggableTask:0x13bfe3d7>):
    300:                             </code>
    301:                         <% else %>
    302:                             <code>
    303:                                 <% working_dir_title = (task.workingDirectory()==nil || task.workingDirectory() == '') ? "Working Directory": "Working Directory: " + task.workingDirectory()  %>
    304:                                 <span class="working_dir" title="<%= working_dir_title %>"><%= "#{task.workingDirectory()}$" %></span>
    305:                                 <span class="command"><%= task.command() %></span>
    306:                                 <span class="arguments"><%= task.arguments() %></span>
  app/views/config_view/templates/show.html.erb:303:in `_app_views_config_view_templates_show_html_erb___2025015285_46419264'
  app/views/config_view/templates/show.html.erb:279:in `_app_views_config_view_templates_show_html_erb___2025015285_46419264'
  app/views/config_view/templates/show.html.erb:140:in `_app_views_config_view_templates_show_html_erb___2025015285_46419264'
  app/views/config_view/templates/show.html.erb:22:in `_app_views_config_view_templates_show_html_erb___2025015285_46419264'
```

**Comment by dsenecalb** [comment]
Hi @arvindsv ,

Today (2 years after this ticket was created), the same error appears.

Please, advice

`2017-01-20 08:55:05,999 FATAL [qtp610952221-139151147] Rails:? -
ActionView::Template::Error (13):
    2:         <% @current_server_health_states.each do |state_in_messages_body| %>
    3:             <% if !state_in_messages_body.isRealSuccess() %>
    4:                 <div class="<%=state_in_messages_body.getLogLevel().name().downcase%>">
    5:                     <div class="message"><%=state_in_messages_body.getMessageWithTimestamp()%></div>
    6:                     <div class="description"><%==state_in_messages_body.getDescription()%></div>
    7:                 </div>
    8:             <% end %>
  sun.util.calendar.BaseCalendar.getCalendarDateFromFixedDate(sun/util/calendar/BaseCalendar.java:454)
  java.util.GregorianCalendar.computeFields(java/util/GregorianCalendar.java:2333)
  java.util.GregorianCalendar.computeFields(java/util/GregorianCalendar.java:2248)
  java.util.Calendar.setTimeInMillis(java/util/Calendar.java:1140)
  java.util.Calendar.setTime(java/util/Calendar.java:1106)
  java.text.SimpleDateFormat.format(java/text/SimpleDateFormat.java:955)
  java.text.SimpleDateFormat.format(java/text/SimpleDateFormat.java:948)
  java.text.DateFormat.format(java/text/DateFormat.java:336)
  com.thoughtworks.go.serverhealth.ServerHealthState.getMessageWithTimestamp(com/thoughtworks/go/serverhealth/ServerHealthState.java:199)
  java.lang.reflect.Method.invoke(java/lang/reflect/Method.java:606)
  ActionView::CompiledTemplates._app_views_server__messages_body_html_erb__272444874_11122(app/views/server/_messages_body.html.erb:5)
  ActionView::CompiledTemplates._app_views_server__messages_body_html_erb__272444874_11122(app/views/server/_messages_body.html.erb:5)
  Java::JavaUtil::Collection.each(file:/gocd/go-server/work/jetty-0.0.0.0-8153-cruise.war-_go-any-/webapp/WEB-INF/lib/jruby-complete-1.7.11.jar!/jruby/java/java_ext/java.util.rb:7)
  Java::JavaUtil::Collection.each(file:/gocd/go-server/work/jetty-0.0.0.0-8153-cruise.war-_go-any-/webapp/WEB-INF/lib/jruby-complete-1.7.11.jar!/jruby/java/java_ext/java.util.rb:7)
  ActionView::CompiledTemplates._app_views_server__messages_body_html_erb__272444874_11122(app/views/server/_messages_body.html.erb:2)
  ActionView::CompiledTemplates._app_views_server__messages_body_html_erb__272444874_11122(app/views/server/_messages_body.html.erb:2)
  ActionView::Template.render(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/template.rb:143)
  ActionView::Template.render(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/template.rb:143)
  Module.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications.rb:161)
  Module.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications.rb:161)
  ActionView::Template.render(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/template.rb:141)
  ActionView::Template.render(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/template.rb:141)
  ActionView::PartialRenderer.render_partial(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/partial_renderer.rb:306)
  ActionView::PartialRenderer.render_partial(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/partial_renderer.rb:306)
  ActionView::PartialRenderer.render(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/partial_renderer.rb:279)
  ActionView::PartialRenderer.render(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/partial_renderer.rb:279)
  ActionView::AbstractRenderer.instrument(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/abstract_renderer.rb:38)
  ActionView::AbstractRenderer.instrument(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/abstract_renderer.rb:38)
  Module.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications.rb:159)
  Module.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications.rb:159)
  ActiveSupport::Notifications::Instrumenter.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications/instrumenter.rb:20)
  ActiveSupport::Notifications::Instrumenter.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications/instrumenter.rb:20)
  ActiveSupport::Notifications::Instrumenter.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications/instrumenter.rb:19)
  ActiveSupport::Notifications::Instrumenter.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications/instrumenter.rb:19)
  Module.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications.rb:159)
  Module.instrument(vendor/bundle/jruby/1.9/gems/activesupport-4.0.4/lib/active_support/notifications.rb:159)
  ActionView::AbstractRenderer.instrument(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/abstract_renderer.rb:38)
  ActionView::AbstractRenderer.instrument(vendor/bundle/jruby/1.9/gems/actionpack-4.0.4/lib/action_view/renderer/abstract_renderer.rb:38)
`

**Comment by arvindsv** [comment]
Fixed in 17.3. Closing.
